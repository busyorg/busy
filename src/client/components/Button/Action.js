import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';
import classNames from 'classnames';
import './Action.less';

const Action = ({ text, loading, disabled, primary, style, small, type, onClick }) => (
  <button
    disabled={disabled}
    style={style}
    className={classNames('Action', {
      'ant-btn-lg': !small,
      'Action--primary': primary,
    })}
    type={type}
    onClick={onClick}
  >
    {loading && <Icon type="loading" />}
    {text}
  </button>
);

Action.propTypes = {
  text: PropTypes.string.isRequired,
  loading: PropTypes.bool,
  disabled: PropTypes.bool,
  primary: PropTypes.bool,
  style: PropTypes.shape(),
  small: PropTypes.bool,
  type: PropTypes.string,
  onClick: PropTypes.func,
};

Action.defaultProps = {
  loading: false,
  disabled: false,
  primary: false,
  style: {},
  small: false,
  type: '',
  onClick: () => {},
};

export default Action;
