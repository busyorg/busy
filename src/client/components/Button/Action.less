@import (reference) '../../styles/custom.less';
@import (reference) '../../styles/mixins.less';

.Action {
  .btn;
  .btn-default;

  .colorize-button(#c2ccd3, #99aab5);

  color: @white;

  &--primary {
    .colorize-button(@primary-color, @primary-button-border);

    color: @white-darker;

    &:disabled,
    &:hover,
    &:focus,
    &:active,
    &.active {
      color: @white-darker !important;
    }
  }

  &--big {
    padding: 0 15px;
    height: 46px;
  }

  &:hover,
  &:focus,
  &:active,
  &.active {
    color: @white;
  }

  & .anticon {
    vertical-align: -1px;
    margin-right: 8px;
  }
}
