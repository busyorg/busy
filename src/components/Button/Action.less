@import "../../styles/custom.less";
@import "../../styles/mixins.less";

.Action {
  .btn;
  .btn-default;

  .colorize-button(#c2ccd3, #99aab5);

  color: @white;
  width: 100%;

  &:hover,
  &:focus,
  &:active,
  &.active {
    color: @white;
  }

  & .anticon {
    vertical-align: -1px;
    margin-right: 8px;
  }
}
