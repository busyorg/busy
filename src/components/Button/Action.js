import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';
import classNames from 'classnames';
import './Action.less';

const Action = ({ text, loading, disabled, primary, style, small, onClick }) => (
  <button
    disabled={disabled}
    style={style}
    className={classNames('Action', {
      'ant-btn-lg': !small,
      'Action--primary': primary,
    })}
    onClick={onClick}
  >
    {loading && <Icon type="loading" />}
    {text}
  </button>
);

Action.propTypes = {
  text: PropTypes.string.isRequired,
  loading: PropTypes.bool,
  disabled: PropTypes.bool,
  primary: PropTypes.bool,
  style: PropTypes.shape(),
  small: PropTypes.bool,
  onClick: PropTypes.func,
};

Action.defaultProps = {
  loading: false,
  disabled: false,
  primary: false,
  style: {},
  small: false,
  onClick: () => {},
};

export default Action;
