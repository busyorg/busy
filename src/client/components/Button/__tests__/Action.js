import React from 'react';
import { shallow } from 'enzyme';
import Action from '../Action';

describe('<Action />', () => {
  it('renders without exploding', () => {
    const props = {
      loading: true,
      disabled: false,
      primary: true,
    };
    const wrapper = shallow(<Action {...props}>Example text</Action>);
    expect(wrapper).toMatchSnapshot();
  });
});
