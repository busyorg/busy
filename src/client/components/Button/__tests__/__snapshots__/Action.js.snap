// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Action
    big={false}
    className=""
    disabled={false}
    loading={true}
    onClick={[Function]}
    primary={true}
  >
    Example text
  </Action>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          type="loading"
        />,
        "Example text",
      ],
      "className": "Action Action--primary",
      "disabled": false,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "type": "loading",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Example text",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            type="loading"
          />,
          "Example text",
        ],
        "className": "Action Action--primary",
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "loading",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Example text",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
